plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -171,21 +197,21 @@ public ConnectorPageSource createPageSource(
         }
 
         ReaderPageSource pageSource = ParquetPageSourceFactory.createPageSource(
-                fileSystemFactory.create(session).newInputFile(split.getPath(), split.getFileSize()),
+                inputFile,
                 split.getStart(),
                 split.getLength(),
-                hiveColumnHandles,
+                hiveColumnHandles.build(),
                 parquetPredicate,
                 true,
                 parquetDateTimeZone,
                 fileFormatDataSourceStats,
-                parquetReaderOptions.withMaxReadBlockSize(getParquetMaxReadBlockSize(session))
-                        .withUseColumnIndex(isParquetUseColumnIndex(session)));
+                options);
 
         verify(pageSource.getReaderColumns().isEmpty(), "All columns expected to be base columns");
 
         return new DeltaLakePageSource(
                 deltaLakeColumns,
+                nullColumnNames.build(),
```

Comment:
I think you could get by without passing the null column names all the way through. The ParquetPageSource should already return nulls for columns whose name doesn't show up in the file. Maybe if the column is missing from the file you could give it a name like `missing_column_<uuid>`? 

I'm not sure if this is cleaner though.

Reply:
Actually, it was my first approach and settled on the current change because I felt it's not straightforward and the `ParquetPageSource` behavior may change in the future.

@findepi Do you have any opinion?

Reply:
I am not being able to fully weight the consequences, but i agree that dummy names look weird.
Would would be the benefit of doing so?

Reply:
It gets rid of the changes in `DeltaLakePageSource`. That's about it

Reply:
I don't really have a preference, happy to merge as is